winpkg/ddev.nsi

```diff
@@ -1,1045 +1,1083 @@
+    SetDetailsView show
+    DetailPrint "Download of `${DOCKER_DESKTOP_NAME}` failed:"
+    DetailPrint " $R0"
+    MessageBox MB_ICONEXCLAMATION|MB_OK "Download of `${DOCKER_DESKTOP_NAME}` has failed, please download and install once this installation has finished. Continue the resting installation."
+  ${EndIf}
+
+  !undef DOCKER_DESKTOP_INSTALLER
+SectionEnd
+!endif ; DOCKER_NSH
+
+/**
+ * sudo application install
+ */
+Section "sudo" SecSudo
+  ; Force installation
+  SectionIn 1 2 3 RO
+  SetOutPath "$INSTDIR"
+  SetOverwrite try
+
+  ; Copy files
+  File "..\.gotmp\bin\windows_amd64\sudo.exe"
+  File "..\.gotmp\bin\windows_amd64\sudo_license.txt"
+SectionEnd
+
+/**
+ * mkcert group
+ */
+SectionGroup /e "mkcert"
+  /**
+   * mkcert application install
+   */
+  Section "mkcert" SecMkcert
+    ; Install in non choco mode only
+    ${IfNot} ${Chocolatey}
+      SectionIn 1 2
+      SetOutPath "$INSTDIR"
+      SetOverwrite try
+
+      ; Copy files
+      File "..\.gotmp\bin\windows_amd64\mkcert.exe"
+      File "..\.gotmp\bin\windows_amd64\mkcert_license.txt"
+
+      ; Install icons
+      SetOutPath "$INSTDIR\Icons"
+      SetOverwrite try
+      File /oname=ca-install.ico "graphics\ca-install.ico"
+      File /oname=ca-uninstall.ico "graphics\ca-uninstall.ico"
+
+      ; Shortcuts
+      CreateShortcut "$INSTDIR\mkcert install.lnk" "$INSTDIR\mkcert.exe" "-install" "$INSTDIR\Icons\ca-install.ico"
+      CreateShortcut "$INSTDIR\mkcert uninstall.lnk" "$INSTDIR\mkcert.exe" "-uninstall" "$INSTDIR\Icons\ca-uninstall.ico"
+
+      !insertmacro MUI_STARTMENU_WRITE_BEGIN Application
+      CreateDirectory "$SMPROGRAMS\$ICONS_GROUP\mkcert"
+      CreateShortCut "$SMPROGRAMS\$ICONS_GROUP\mkcert\mkcert install trusted https.lnk" "$INSTDIR\mkcert install.lnk"
+      CreateShortCut "$SMPROGRAMS\$ICONS_GROUP\mkcert\mkcert uninstall trusted https.lnk" "$INSTDIR\mkcert uninstall.lnk"
+      !insertmacro MUI_STARTMENU_WRITE_END
+    ${EndIf}
+  SectionEnd
+
+  /**
+   * mkcert setup
+   */
+  Section "Setup mkcert" SecMkcertSetup
+    ; Install in non silent and choco mode only
+    ${IfNot} ${Silent}
+    ${AndIfNot} ${Chocolatey}
+      MessageBox MB_ICONINFORMATION|MB_OK "Now running mkcert to enable trusted https. Please accept the mkcert dialog box that may follow."
+
+      ; Run setup
+      nsExec::ExecToLog '"$INSTDIR\mkcert.exe" -install'
+      Pop $R0 ; get return value
+
+      ; Check return value and write setup status to registry on success
+      ${If} $R0 = 0
+        WriteRegDWORD ${REG_UNINST_ROOT} "${REG_UNINST_KEY}" "NSIS:mkcertSetup" 1
+      ${EndIf}
+    ${EndIf}
+  SectionEnd
+SectionGroupEnd
+
+/**
+ * WinNFSd group
+ */
+SectionGroup /e "WinNFSd"
+  /**
+   * WinNFSd application install
+   */
+  Section "WinNFSd" SecWinNFSd
+    SectionIn 1
+    SetOutPath "$INSTDIR"
+    SetOverwrite try
+
+    ; Copy files
+    ;File "..\.gotmp\bin\windows_amd64\winnfsd.exe"
+    File "..\.gotmp\bin\windows_amd64\winnfsd_license.txt"
```

Comment:
This file is currently downloaded by the Makefile, it's OK, but a little odd since we get the binary while running the thing. Probably slightly better to remove from the Makefile and download with the binary.